tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AspNetCore/Mvc/DefaultApplicationModelProviderCreateActionModelIntegration.cs

```diff
@@ -0,0 +1,86 @@
+// <copyright file="DefaultApplicationModelProviderCreateActionModelIntegration.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+#nullable enable
+
+#if !NETFRAMEWORK
+
+using System;
+using System.ComponentModel;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using Datadog.Trace.ClrProfiler.CallTarget;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.DuckTyping;
+using Datadog.Trace.Iast;
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.AspNetCore.Mvc;
+
+/// <summary>
+/// Microsoft.AspNetCore.Mvc.ApplicationModels.ActionModel Microsoft.AspNetCore.Mvc.ApplicationModels.DefaultApplicationModelProvider::CreateActionModel(System.Reflection.TypeInfo,System.Reflection.MethodInfo) calltarget instrumentation
+/// </summary>
+[InstrumentMethod(
+    AssemblyName = "Microsoft.AspNetCore.Mvc.Core",
+    TypeName = "Microsoft.AspNetCore.Mvc.ApplicationModels.DefaultApplicationModelProvider",
+    MethodName = "CreateActionModel",
+    ReturnTypeName = "Microsoft.AspNetCore.Mvc.ApplicationModels.ActionModel",
+    ParameterTypeNames = ["System.Reflection.TypeInfo", "System.Reflection.MethodInfo"],
+    MinimumVersion = "2.0.0.0",
+    MaximumVersion = "8.*.*.*.*",
+    IntegrationName = nameof(IntegrationId.AspNetCore))]
+[Browsable(false)]
+[EditorBrowsable(EditorBrowsableState.Never)]
+public class DefaultApplicationModelProviderCreateActionModelIntegration
+{
+    private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor<DefaultApplicationModelProviderCreateActionModelIntegration>();
+    private static readonly string[] AllowedVerbs = ["GET", "POST"];
+
+    internal static CallTargetReturn<TReturn?> OnMethodEnd<TTarget, TReturn>(TTarget instance, TReturn? returnValue, Exception? exception, in CallTargetState state)
+    {
+        try
+        {
+            AnalyzeActionModel(returnValue);
+        }
+        catch (Exception ex)
+        {
+            Log.Warning(ex, "Failed to analyze or report Verb Tampering vulnerability");
+        }
+
+        return new CallTargetReturn<TReturn?>(returnValue);
+    }
+
+    private static void AnalyzeActionModel(object? returnValue)
+    {
+        if (returnValue == null) { return; }
+        if (!returnValue.TryDuckCast<IActionModel>(out var actionModel)) { return; }
```

Comment:
It would be better to do this duck typing using constraints on the `TReturn` type?